packages/flutter/test/widgets/reorderable_list_test.dart

```diff
@@ -923,6 +923,52 @@ void main() {
     ), throwsAssertionError);
   });
 
+  testWidgets('ReorderableList passes itemExtentBuilder to SliverReorderableList', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/155936
+    const int itemCount = 5;
+    const List<double> items = <double>[10.0, 20.0, 30.0, 40.0, 50.0];
+    final Map<int, double> itemExtents = <int, double>{};
+
+    void handleReorder(int fromIndex, int toIndex) {
+      if (toIndex > fromIndex) {
+        toIndex -= 1;
+      }
+      items.insert(toIndex, items.removeAt(fromIndex));
+    }
+
+    // The list has five elements, that indicate the extent for the item at the given index.
+    await tester.pumpWidget(
+      MaterialApp(
+        home: ReorderableList(
+          itemBuilder: (BuildContext context, int index) => SizedBox(
+            key: ValueKey<double>(items[index]),
+            width: 100,
+            height: items[index],
+          ),
+          itemCount: itemCount,
+          onReorder: handleReorder,
+          itemExtentBuilder: (int index, SliverLayoutDimensions dimensions) {
+            final double extent = items[index];
+
+            itemExtents[index] = extent;
+
+            return extent;
+          },
+        ),
+      ),
+    );
+
+    const Map<int, double> expectedExtents = <int, double>{
+      0: 10.0,
+      1: 20.0,
+      2: 30.0,
+      3: 40.0,
+      4: 50.0,
+    };
+
```

Comment:
I think getting the size for each child and comparing them would be more straightforward:) Like this(I modified your test a little bit):

```dart
testWidgets('ReorderableList passes itemExtentBuilder to SliverReorderableList', (WidgetTester tester) async {
    // Regression test for https://github.com/flutter/flutter/issues/155936
    const int itemCount = 5;
    const List<double> items = <double>[10.0, 20.0, 30.0, 40.0, 50.0];
    final Map<int, double> itemExtents = <int, double>{};

    void handleReorder(int fromIndex, int toIndex) {
      if (toIndex > fromIndex) {
        toIndex -= 1;
      }
      items.insert(toIndex, items.removeAt(fromIndex));
    }

    // The list has five elements, that indicate the extent for the item at the given index.
    await tester.pumpWidget(
      MaterialApp(
        home: ReorderableList(
          itemBuilder: (BuildContext context, int index) => SizedBox(
            key: ValueKey<double>(items[index]),
            child: Text('Item $index'),
          ),
          itemCount: itemCount,
          onReorder: handleReorder,
          itemExtentBuilder: (int index, SliverLayoutDimensions dimensions) {
            return items[index];
          },
        ),
      ),
    );

    const Map<int, double> expectedExtents = <int, double>{
      0: 10.0,
      1: 20.0,
      2: 30.0,
      3: 40.0,
      4: 50.0,
    };

    for (int i = 0; i < itemCount; i++) {
      itemExtents[i] = tester.getSize(find.text('Item $i')).height;
    }
    expect(const MapEquality<int, double>().equals(itemExtents, expectedExtents), isTrue);
  });
```

Reply:
Done. I also moved the declaration of `itemExtents` to just before the expect, as we can initialize it using a for-literal, using your suggestion as the body for the literal :)